cdap-validation/bin/testframework.py

```diff
@@ -0,0 +1,262 @@
+
+
+def test_api_connection(base_info, cluster_info, api_tests):
+    if cluster_info['verbose'] == 2: print 'test api connection'
+    # let us iterate through potential hadoop manager tests and see what we get back
+    # the first one that works should be saved and set from that point on for the rest of the testing
+    host_url = cluster_info['host']
+    for hash in api_tests:
+        if cluster_info['verbose'] == 2: print ''
+        if cluster_info['verbose'] == 2: print hash
+        for manager, test in hash.iteritems():
+            mgr_test_url = host_url + test
+            # print "manager=%s, test=%s, mgr_test_url=%s" % (manager, test, mgr_test_url)
+            if cluster_info['verbose'] == 2: print 'Running onetime_auth'
+            helpers.onetime_auth(mgr_test_url, cluster_info) # set up password manager and install opener
+            if cluster_info['verbose'] == 2: print 'run_request'
+            h = helpers.run_request(mgr_test_url, cluster_info) # run url request
+            # if it equals noapi, it means the api login test failed for that install manager
+            if h == 'noapi':
+                if cluster_info['verbose'] == 2: print 'No API for %s, trying the next or exiting' % (manager)
+                break
+            mgr_test = h.read()
+            # add logic to do next iteration or not, depending on whether we succeeded on first one
+            # if good: create array or hash with manager and mgr_test and return that?
+            if mgr_test == 'v10':
+                return mgr_test
+                break
+        else:
+            break
+
+
+def get_install_manager(version):
+    if version == 'v10':
+        manager = 'cloudera'
+    else:
+        manager = 'ambari'
+
+    return manager
+
+
+# get static info needed to connect to a specific API
+def get_install_manager_info(cluster_info):
+    my_cluster = '/clusters/' + cluster_info['cluster']
+    cluster_info['my_cluster'] = my_cluster
+    base_url = cluster_info['host']
+    base_url += cluster_info['version']
+    if cluster_info['verbose'] == 2: print 'base_url=%s' % (base_url)
+    cluster_info['base_url'] = base_url
+    return cluster_info
+
+
+# get (and run) API commands
+def get_and_run_api_commands(base, cluster):
+    if cluster['verbose'] == 2: print 'Now running API commands'
+    mgr = cluster['manager']
+    host_url = cluster['base_url']
+    username = cluster['username']
+    password = cluster['password']
+    configs_subdir = base[mgr]['subdir']
+    my_cluster = cluster['my_cluster']
+
+    if mgr == 'cloudera':
+        print 'Hadoop install manager: Cloudera Manager'
+        #cloudera.cloudera_commands(host_url, configs_subdir, base, cluster) ## disabling for now
+
+    elif mgr == 'ambari':
+        if cluster['verbose'] == 2: print 'Hadoop install manager: Ambari'
+        ambari.get_ambari_configs(host_url, configs_subdir, base, cluster)
+
+    else:
+        print 'Your Hadoop install manager is not recognized'
```

Comment:
I would put this in it's own method... `get_config_from_manager(mgr, host_url, configs_subdir, base, cluster)` and put this conditional in that method.